crates/masonry/src/widget/widget_mut.rs

```diff
@@ -21,67 +18,34 @@ use crate::{Widget, WidgetId, WidgetState};
 ///
 /// `WidgetMut` implements [`Deref`] with `W::Mut` as target.
 ///
-/// ## Internals
-///
-/// `WidgetMut<W>` requires that W implement [`StoreInWidgetMut`]; it stores a `W::Mut`,
-/// which is a special type declared with the `declare_widget` macro. Methods to mutate
-/// the widget will be implemented in that `W::Mut` type, which `WidgetMut<W>` derefs to.
+/// ## WidgetMut as a Receiver
 ///
-/// See [`declare_widget`](crate::declare_widget) for details.
-pub struct WidgetMut<'a, W: Widget + StoreInWidgetMut> {
-    pub(crate) parent_widget_state: &'a mut WidgetState,
-    pub(crate) inner: W::Mut<'a>,
+/// Once the Receiver trait is stabilized, WidgetMut will implement it so that custom
+/// widgets in downstream crates can use WidgetMut as the receiver for inherent methods.
```

Comment:
I'd at least mention extension traits here

Reply:
Right, this is very minimal documentation. I should probably have added a TODO.